src/drivers/net/mod.rs

```diff
@@ -5,22 +5,43 @@
 // http://opensource.org/licenses/MIT>, at your option. This file may not be
 // copied, modified, or distributed except according to those terms.
 
+use crate::arch::irq;
 use crate::arch::kernel::percore::*;
 use crate::scheduler::task::TaskHandle;
 use crate::synch::semaphore::*;
 use crate::synch::spinlock::SpinlockIrqSave;
 use alloc::collections::BTreeMap;
+use core::sync::atomic::{AtomicBool, Ordering};
 
 static NET_SEM: Semaphore = Semaphore::new(0);
 static NIC_QUEUE: SpinlockIrqSave<BTreeMap<usize, TaskHandle>> =
 	SpinlockIrqSave::new(BTreeMap::new());
+static POLLING: AtomicBool = AtomicBool::new(false);
+
+/// set driver in polling mode and threads will not be blocked
+fn set_polling_mode(value: bool) {
+	// is the driver already in polling mode?
+	if POLLING.swap(value, Ordering::SeqCst) != value {
+		let irq = irq::nested_disable();
+
+		if let Some(driver) = crate::arch::kernel::pci::get_network_driver() {
+			driver.borrow_mut().set_polling_mode(value);
+		}
+
+		irq::nested_enable(irq);
```

Comment:
Can you use the function taking the closure from the other pull request?

Reply:
In this case, #91 must be accepted.

Reply:
The last commit solve this problem. Now, I protect the network driver by a lock.